src/config.js

```diff
@@ -0,0 +1,1602 @@
+      const scopedDefaults = {}
+      for (let scope in schema.scopes) {
+        const scopeSchema = schema.scopes[scope]
+        if (!scopeSchema.hasOwnProperty('default')) { continue }
+        scopedDefaults[scope] = {}
+        setValueAtKeyPath(scopedDefaults[scope], keyPath, scopeSchema.default)
+      }
+      this.scopedSettingsStore.addProperties('schema-default', scopedDefaults)
+    }
+
+    if ((schema.type === 'object') && (schema.properties != null) && isPlainObject(schema.properties)) {
+      const keys = splitKeyPath(keyPath)
+      for (let key in schema.properties) {
+        const childValue = schema.properties[key]
+        if (!schema.properties.hasOwnProperty(key)) { continue }
+        this.setScopedDefaultsFromSchema(keys.concat([key]).join('.'), childValue)
+      }
+    }
+  }
+
+  extractDefaultsFromSchema (schema) {
+    if (schema.default != null) {
+      return schema.default
+    } else if ((schema.type === 'object') && (schema.properties != null) && isPlainObject(schema.properties)) {
+      const defaults = {}
+      const properties = schema.properties || {}
+      for (let key in properties) { const value = properties[key]; defaults[key] = this.extractDefaultsFromSchema(value) }
+      return defaults
+    }
+  }
+
+  makeValueConformToSchema (keyPath, value, options) {
+    if (options != null ? options.suppressException : undefined) {
+      try {
+        return this.makeValueConformToSchema(keyPath, value)
+      } catch (e) {
+        return undefined
+      }
+    } else {
+      let schema
+      if ((schema = this.getSchema(keyPath)) == null) {
+        if (schema === false) { throw new Error(`Illegal key path ${keyPath}`) }
+      }
+      return this.constructor.executeSchemaEnforcers(keyPath, value, schema)
+    }
+  }
+
+  // When the schema is changed / added, there may be values set in the config
+  // that do not conform to the schema. This will reset make them conform.
+  resetSettingsForSchemaChange (source) {
+    if (source == null) { source = this.getUserConfigPath() }
+    return this.transact(() => {
+      this.settings = this.makeValueConformToSchema(null, this.settings, {suppressException: true})
+      const selectorsAndSettings = this.scopedSettingsStore.propertiesForSource(source)
+      this.scopedSettingsStore.removePropertiesForSource(source)
+      for (let scopeSelector in selectorsAndSettings) {
+        let settings = selectorsAndSettings[scopeSelector]
+        settings = this.makeValueConformToSchema(null, settings, {suppressException: true})
+        this.setRawScopedValue(null, settings, source, scopeSelector)
+      }
+    })
+  }
+
+  /*
+  Section: Private Scoped Settings
+  */
+
+  priorityForSource (source) {
+    if (source === this.getUserConfigPath()) {
+      return 1000
+    } else {
+      return 0
+    }
+  }
+
+  emitChangeEvent () {
+    if (!(this.transactDepth > 0)) { return this.emitter.emit('did-change') }
```

Comment:
looks like a decaffeinated `unless`? assuming `transactDepth` is always a number, you can remove the negation and do `<=`.